### PR TITLE
feat(PT-2815): reservoir orders reconciliation

### PR DESCRIPTION
**Pull request by dmitry-saraykin**

### CODE REVIEW

File: order/listener/src/main/kotlin/com/rarible/protocol/order/listener/configuration/OrderListenerConfiguration.kt [hunk 1]
```diff
@@ -335,4 +339,30 @@ class OrderListenerConfiguration(
             handler = looksrareOrderLoadHandler
         ).apply { start() }
     }
+
+    @Bean
```
Comment (yakischik) [inline order/listener/src/main/kotlin/com/rarible/protocol/order/listener/configuration/OrderListenerConfiguration.kt:343]: Why not in worker's class?

File: order/listener/src/main/kotlin/com/rarible/protocol/order/listener/job/ReservoirOrdersReconciliationWorker.kt [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+    private suspend fun recordState(response: ReservoirOrderEvents) {
+        response.continuation?.run {
+            aggregatorStateRepository.save(ReservoirAsksEventFetchState(cursor = this))
+        }
+    }
+
+    private suspend fun waitIfNecessary(response: ReservoirOrderEvents) {
+        if (response.events.size < properties.size) {
+            logger.info("Result is not full. Will wait for ${properties.pollingPeriod}")
```
Comment (yakischik) [inline order/listener/src/main/kotlin/com/rarible/protocol/order/listener/job/ReservoirOrdersReconciliationWorker.kt:76]: Better to log some data here like actual size vs expected size

File: order/listener/src/main/kotlin/com/rarible/protocol/order/listener/misc/ForeignOrderMetrics.kt [hunk 3]
```diff
@@ -100,11 +100,23 @@ class ForeignOrderMetrics(
         ).increment()
     }
 
+    fun onOrderInconsistency(
+        platform: Platform,
+    ) {
+        meterRegistry.counter(
+            FOREIGN_ORDER_INCONSISTENCY,
```
Comment (yakischik) [inline order/listener/src/main/kotlin/com/rarible/protocol/order/listener/misc/ForeignOrderMetrics.kt:107]: Maybe it make sense to add tag "status"?

**Review by yakischik — APPROVED** [review]

File: order/listener/src/main/kotlin/com/rarible/protocol/order/listener/configuration/OrderListenerConfiguration.kt (hunk 1 above)
Comment (dmitry-saraykin) [reply order/listener/src/main/kotlin/com/rarible/protocol/order/listener/configuration/OrderListenerConfiguration.kt:343]: for consistency with other workers

**Review by yakischik — APPROVED** [review]

**Pull request merged**